docs/data.fetch.md

```diff
@@ -0,0 +1,82 @@
+# qgrid filtering interface
+
+Filtering is tightly coupled with ordering, paging and virtualization.
+
+Data fetch process is going to be a pipline of ordered operations:
+
+ 1. **Fetch** - returns a promise which returns data.
+ 2. **Filter** (optional) - filters data according to `match` function. Gets data from `Fetch` stage.
+ 3. **Order** (optional) - orders data according to `comparator` function provided by user. Gets data from `Filter` stage. 
+ 4. **Page** (optional) - defines current page and page size  
+ 5. **Virtualize** (optional) - defines number of rendered items.
+
+Every operation gets data from previous stage and handles it properly.
+If user wants to perform some of this actions on client-side, he modifies pipeline by adding a function to corresponding model.
+
+## Common cases
+1. All data is filtered and ordered on server-side.
+```javascript
+gridModel
+    .data({
+        fetch: (model) => {
+            return $http({
+                filter: model.filter,
+                order: model.filter
+            })
+        }
+    });
+```
+
+2. Data is fetched from server but filtering, ordering and pagination is performed on server.
+
+```javascript
+gridModel
+    .data({
+        fetch: (model) => {
+            return $http();
+        }
+    })
+    .filter({
+        match: (model, row) => {
+            // some filtering logic here
```

Comment:
as an idea we can make interface like collection pipeline for client side
```javascript
.data({
   pipeline/pipe: [
      (data, model) => do filter,
      (data, model) => do grouping
      (data, model) => do ordering
      ....
   ]
})

however it will be harder to change just one pipe element (e.g. just filtering) without touching ordering,
in other hand it looks extensible